src/store/actions/api.actions.js

```diff
@@ -103,17 +197,79 @@ export const deleteEndpoint = (apiName, callback) => async (dispatch) => {
     }
 };
 
-export const fetchEndpoint = pathname => (dispatch) => {
+export const fetchEndpoint = pathname => async (dispatch) => {
     dispatch(getEndpointRequest());
 
-    return client.get(`apis${pathname}`)
-        .then((response) => {
-            dispatch(getEndpointSuccess(response.data));
-        })
-        .catch((error) => {
-            // eslint-disable-next-line
-            console.log('FETCH_ENDPOINT_ERROR', 'Infernal server error', error);
+    try {
+        const response = await client.get(`apis${pathname}`);
+        const preparedPlugins = response.data.plugins.map(plugin => {
```

Comment:
If this function is the same of the above in `willClone` method just extract this ad reuse it in both places

Reply:
almost, but has some differences. I'll try to find the way to extract the common part in separate **PR**